python/dgl/distributed/optim/pytorch/sparse_optim.py

```diff
@@ -35,16 +45,168 @@ def __init__(self, params, lr):
             self._rank = 0
             self._world_size = 1
 
+    def local_state_dict(self):
+        """Return the state pertaining to current rank of the optimizer.
+
+        Returns
+        -------
+        dict
+            Local state dict
+
+        See Also
+        --------
+        load_local_state_dict
+        """
+        lcoal_state_dict = {}
+        lcoal_state_dict["emb_states"] = {}
+        num_clients = dgl.distributed.get_num_client()
+        lcoal_state_dict["params"] = {"num_clients": num_clients}
+        for emb in self._params:
+            kvstore = emb._tensor.kvstore
+            trainers_per_machine = (
+                num_clients // dgl.distributed.get_num_machines()
+            )
+            emb_state_dict = {}
+            idx_i = F.tensor(range(emb.weight.shape[0]), F.int64)
+            part_policy = (
+                emb.part_policy if emb.part_policy else emb._tensor.part_policy
+            )
+            part_id = kvstore.get_partid(emb.data_name, idx_i)
+            mask = part_id == part_policy.part_id
+            idx_i = F.boolean_mask(idx_i, mask)
+            if trainers_per_machine > 1:
```

Comment:
below logic is about how to dispatch workload between trainers on same machine: save a range of embeddings into each trainer. Right?
is it possible to extract such logic so that we may reuse it for `edge embeddings` in the future? 

The key part here is to extract a subtensor of `DistTensor` and save it, right?

Reply:
Yes, but I'm not sure it is a standard usage for dispatch, so how about we design it more general when needed?